### PR TITLE
GeoJSON, orgunits and error messages

### PR DESCRIPTION
**Pull request by hermantretteteig**
- Added back GeoJSON/polygon
- Only exporting needed orgUnits
- Display http error messages
- Only possible to export data for one level
- Changed some files from jsx to tsx

### CODE REVIEW

File: src/components/prediction/DownloadData.tsx [hunk 1]
```diff
@@ -42,20 +49,31 @@ const DownloadData = ({ period, setStartDownload, orgUnitLevels, orgUnits, tempe
...
+  const {data : population, error : populationError, loading : populationLoading} = useAnalyticRequest(populationData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const {data : temperature, error : temperatureError, loading : temperatureLoading} = useAnalyticRequest(temperatureData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const {data : prediction, error : predictionError,loading : predictionLoading} = useAnalyticRequest(predictionData.id, fillPeriodesMonths(period), mergedOrgUnits)
+  const {data : geoJson, error : geoJsonError, loading : geoJsonLoading} = useGeoJson(orgUnitLevel.level);
 
   const objectToPrettyJson = (object : any) => {
     return JSON.stringify(object, null, 2)
   }
 
+  //Avoid to export all the orgUnits to JSON, only inlcude those returned from one of the analytic requests (precipitation)
```
Comment (hermantretteteig) [inline src/components/prediction/DownloadData.tsx:62]: Did not found any way to filter on this in the API-request..

File: src/components/prediction/PredictionPage.tsx [hunk 2]
```diff
@@ -79,10 +98,20 @@ const PredictionPage = () => {
...
       )}
+      {errorMessages.map((error : ErrorResponse, index) => (
+        <div key={index} className={styles.errorBar}>
+          <div className={styles.errorHeader}>
+            <IconError24/>
+            <span>{error.element} request failed</span>
+          </div>
+            <span className={styles.detailedError}>{JSON.stringify(error.error, null, 2)}</span>
+        </div>
+      ))}
```
Comment (hermantretteteig) [inline src/components/prediction/PredictionPage.tsx:114]: This looks like this
image

File: src/components/prediction/DataElement.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import PropTypes from "prop-types";
 import i18n from "@dhis2/d2-i18n";
 import { useDataQuery } from "@dhis2/app-runtime";
 import { SingleSelectField, SingleSelectOption } from "@dhis2/ui";
+import React from "react";
```
Comment (turban) [inline src/components/prediction/DataElement.tsx:6]: Better to include with useEffect on line 1:

`import React, { useEffect } from "react";`

File: src/components/prediction/DownloadData.tsx [hunk 4]
```diff
@@ -5,22 +5,29 @@ import { saveAs } from "file-saver";
 import useAnalyticRequest from "../../hooks/useAnalyticRequest";
 import useOrgUnits from "../../hooks/useOrgUnits";
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:6]: `useOrgUnits` is no longer in use in this file

File: src/components/prediction/DownloadData.tsx [hunk 5]
```diff
@@ -5,22 +5,29 @@ import { saveAs } from "file-saver";
...
 import React from "react";
+import useGeoJson from "../../hooks/useGeoJson";
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:9]: `PropTypes` import is not used in this file

File: src/components/prediction/DownloadData.tsx [hunk 6]
```diff
@@ -42,20 +49,31 @@ const DownloadData = ({ period, setStartDownload, orgUnitLevels, orgUnits, tempe
...
+    }
+  }
+
+
   const downloadZip = ()  => {
     const zip = new JSZip();
-    //Add orgUnits to zip
-    zip.file("orgUnits.json", objectToPrettyJson(allOrgUnits))
+    //Add data to zip
+    zip.file("geoJson.json", objectToPrettyJson(filterOrgUnits()))
```
Comment (turban) [inline src/components/prediction/DownloadData.tsx:76]: Maybe call the file `orgUnits.geojson?` .geojson is a recognized in QGIS etc.

File: src/components/prediction/OrgUnitTree.tsx [hunk 7]
```diff
@@ -31,7 +31,7 @@ const OrgUnitTree = ({ selectedOrgUnits, onChange } : OrgUnitTreeProps) => {
         roots={roots.map((r : any) => r.id)}
         selected={selectedOrgUnits?.map((e : any) => e.path)}
         onChange={onChange}
-      
+
```
Comment (turban) [inline src/components/prediction/OrgUnitTree.tsx:34]: Minor: Remove newline

File: src/hooks/useGeoJson.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  })),
+  ...data,}
+}
+
+const useGeoJson = (level : number) => {
+
+    const { data, loading, error } = useDataQuery(ORG_UNITS_QUERY as any, {
+      variables: { level }});
+
+    console.log(loading);
```
Comment (turban) [inline src/hooks/useGeoJson.tsx:29]: Remove `console.log`

**Review by turban — APPROVED** [review]
Approved with some comments that should be fixed before merging.

**Pull request merged**